src/Model/CodecCursor.php

```diff
@@ -73,9 +76,23 @@ public static function fromCursor(Cursor $cursor, DocumentCodec $codec): self
         return new self($cursor, $codec);
     }
 
-    public function getId(): CursorId
+    /** @return CursorId|Int64 */
+    #[ReturnTypeWillChange]
+    public function getId(bool $asInt64 = false)
```

Comment:
For BC layer, adding an argument should be virtual (`func_get_arg`).

Reply:
I thought the argument with a default value would be fine, can you explain why `func_get_args` works better?

Reply:
Although I'm not sure why or how anyone would extend this class.